subprojects/docs/src/docs/userguide/jvm/jvm_test_suite_plugin.adoc

```diff
@@ -0,0 +1,165 @@
+// Copyright 2021 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[jvm_test_suite_plugin]]
+= The JVM Test Suite Plugin
+
+The JVM Test Suite plugin (plugin id: `jvm-test-suite`) allows for modeling multiple groups of automated tests in a Java project.  Tests suites are intended to grouped by similarities between the tests, e.g. fast tests vs. slow tests, or grouped by their common testing framework.
+
+This plugin can be used to define a group of Integration Tests, which might run much longer than unit tests and have different environmental requirements.
+
+[[sec:jvm_test_suite_usage]]
+== Usage
+
+This plugin is applied automatically by the `java` plugin, and can be additionally applied explicitly if desired:
+
+.Using the JVM Test Suite plugin
+====
+include::sample[dir="snippets/testing/test-suite-plugin/groovy",files="build.gradle[tags=apply-jvm-test-suite]"]
+include::sample[dir="snippets/testing/test-suite-plugin/kotlin",files="build.gradle.kts[tags=apply-jvm-test-suite]"]
+====
+
+The plugin cannot be applied standalone as it relies on several conventions of the `java` plugin.
+
+The plugins adds the following objects to the project:
+
+ * A `testing` extension (type: link:{groovyDslPath}/org.gradle.testing.base.TestingExtension.html[TestingExtension]) to the project used to configure test suites.
+ * A test suite named `test` (type: link:{groovyDslPath}/org.gradle.api.plugins.jvm.JvmTestSuite.html[JvmTestSuite]).
+ * A `test` link:{groovyDslPath}/org.gradle.api.tasks.SourceSet.html[SourceSet].
+ * Several configurations derived from the `test` SourceSet name: `testImplementation`, `testCompileOnly`, `testRuntimeOnly`
```

Comment:
```suggestion
 * Several configurations derived from the `test` SourceSet named: `testImplementation`, `testCompileOnly`, `testRuntimeOnly`
```

Reply:
I think this is correct as-is, or how about:

> Several configurations are derived from the name of the `test` SourceSet: ...

Reply:
that's fine